lib/graphql/object_type.rb

```diff
@@ -38,35 +38,31 @@ def initialize
       super
       @fields = {}
       @interface_fields = {}
-      @dirty_interfaces = []
-      @dirty_inherited_interfaces = []
+      @interface_type_memberships = []
+      @inherited_interface_type_memberships = []
     end
 
     def initialize_copy(other)
       super
-      @clean_interfaces = nil
-      @clean_inherited_interfaces = nil
-      @dirty_interfaces = other.dirty_interfaces.dup
-      @dirty_inherited_interfaces = other.dirty_inherited_interfaces.dup
+      @interface_type_memberships = other.interface_type_memberships.dup
+      @inherited_interface_type_memberships = other.inherited_interface_type_memberships.dup
       @fields = other.fields.dup
     end
 
     # This method declares interfaces for this type AND inherits any field definitions
     # @param new_interfaces [Array<GraphQL::Interface>] interfaces that this type implements
     # @deprecated Use `implements` instead of `interfaces`.
     def interfaces=(new_interfaces)
-      @clean_interfaces = nil
-      @clean_inherited_interfaces = nil
-      @clean_inherited_fields = nil
-
-      @dirty_inherited_interfaces = []
+      @interface_type_memberships = []
+      @inherited_interface_type_memberships = []
       @dirty_inherited_fields = {}
+      @clean_inherited_fields = nil
+      @cached_interfaces = nil
       implements(new_interfaces, inherit: true)
     end
 
-    def interfaces
-      load_interfaces
-      @clean_interfaces
+    def interfaces(ctx = GraphQL::Query::NullContext)
+      load_interfaces(ctx)
```

Comment:
I used to have a `@cached_interfaces` variable here which would cache interfaces on a null context, but it was leading to a ton of flakey tests. I can dig into it more as to why caching wasn't working.

The error I was getting was that `Cheese` (a GraphQL Object type) was being referenced as a interface on itself... but it isn't an interface so I am not sure how it was even getting assigned as an interface on it's own object.